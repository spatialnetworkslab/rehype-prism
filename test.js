'use strict';

const rehype = require('rehype');
const dedent = require('dedent');
const rehypePrism = require('./index');
const svelte = require('@snlab/refractor-svelte');

const processHtml = (html, options) => {
  return rehype()
    .data('settings', { fragment: true })
    .use(rehypePrism, options)
    .processSync(html)
    .toString();
};

test('copies the language- class to pre tag', () => {
  const result = processHtml(dedent`
    <pre><code class="language-css"></code></pre>
  `);
  expect(result).toMatchSnapshot();
});

test('finds code and highlights', () => {
  const result = processHtml(dedent`
    <div>
      <p>foo</p>
      <pre><code class="language-css">p { color: red }</code></pre>
    </div>
  `);
  expect(result).toMatchSnapshot();
});

test('handles uppercase languages correctly', () => {
  const result = processHtml(dedent`
    <div>
      <p>foo</p>
      <pre><code class="language-CSS">p { color: red }</code></pre>
    </div>
  `);
  expect(result).toMatchSnapshot();
});

test('does nothing to code block without language- class', () => {
  const result = processHtml(dedent`
    <pre><code>p { color: red }</code></pre>
  `);
  expect(result).toMatchSnapshot();
});

test('throw error with fake language- class', () => {
  expect(() => {
    processHtml(dedent`
      <pre><code class="language-thisisnotalanguage">p { color: red }</code></pre>
    `);
  }).toThrow(/Unknown language/);
});

test('with options.ignoreMissing, does nothing to code block with fake language- class', () => {
  const html = dedent`
    <pre><code class="language-thisisnotalanguage">p { color: red }</code></pre>
  `;
  const result = processHtml(html, { ignoreMissing: true });
  expect(result).toMatchSnapshot();
});

test('throw error with wrongly specified options.registerSyntax', () => {
  expect(() => {
    processHtml(
      dedent`
    <pre><code class="language-css"></code></pre>
    `,
      { registerSyntax: true }
    );
  }).toThrow(/should be an array/);
});

test('additional language syntax gets registered and applied correctly', () => {
  const html = dedent`
  <pre><code class="language-svelte">
    {#each items as item, i}
      {item.name}
    {/each}
  </code></pre>
  `;
  const result = processHtml(html, { registerSyntax: [svelte] });
  expect(result).toMatchSnapshot();
});
