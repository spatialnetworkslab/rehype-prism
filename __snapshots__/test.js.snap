// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`additional language syntax gets registered and applied correctly 1`] = `
"<pre class=\\"language-svelte\\"><code class=\\"language-svelte\\">
  <span class=\\"token each\\"><span class=\\"token punctuation\\">{</span><span class=\\"token keyword\\">#each</span> <span class=\\"token language-javascript\\">items </span><span class=\\"token keyword\\">as</span> <span class=\\"token language-javascript\\">item<span class=\\"token punctuation\\">,</span> i<span class=\\"token punctuation\\">}</span></span></span>
    <span class=\\"token language-javascript\\"><span class=\\"token punctuation\\">{</span>item<span class=\\"token punctuation\\">.</span><span class=\\"token property-access\\">name</span><span class=\\"token punctuation\\">}</span></span>
  <span class=\\"token language-javascript\\"><span class=\\"token punctuation\\">{</span><span class=\\"token operator\\">/</span>each<span class=\\"token punctuation\\">}</span></span>
</code></pre>"
`;

exports[`copies the language- class to pre tag 1`] = `"<pre class=\\"language-css\\"><code class=\\"language-css\\"></code></pre>"`;

exports[`does nothing to code block without language- class 1`] = `"<pre><code>p { color: red }</code></pre>"`;

exports[`finds code and highlights 1`] = `
"<div>
  <p>foo</p>
  <pre class=\\"language-css\\"><code class=\\"language-css\\"><span class=\\"token selector\\">p</span> <span class=\\"token punctuation\\">{</span> <span class=\\"token property\\">color</span><span class=\\"token punctuation\\">:</span> red <span class=\\"token punctuation\\">}</span></code></pre>
</div>"
`;

exports[`handles uppercase languages correctly 1`] = `
"<div>
  <p>foo</p>
  <pre class=\\"language-css\\"><code class=\\"language-CSS\\"><span class=\\"token selector\\">p</span> <span class=\\"token punctuation\\">{</span> <span class=\\"token property\\">color</span><span class=\\"token punctuation\\">:</span> red <span class=\\"token punctuation\\">}</span></code></pre>
</div>"
`;

exports[`with options.ignoreMissing, does nothing to code block with fake language- class 1`] = `"<pre class=\\"language-thisisnotalanguage\\"><code class=\\"language-thisisnotalanguage\\">p { color: red }</code></pre>"`;
